src/Compilers/CSharp/Portable/BoundTree/NullabilityRewriter.cs

```diff
@@ -74,18 +74,114 @@ private BoundNode VisitBinaryOperatorBase(BoundBinaryOperatorBase binaryOperator
+
+                case LocalSymbol local:
+                    return (T)remapLocal(local);
+
+                case ParameterSymbol param:
+                    if (_remappedSymbols.TryGetValue(param, out var updatedParam))
+                    {
+                        return (T)updatedParam;
+                    }
+                    break;
             }
 
-            return sym;
+            return (T)updatedSymbol;
+
+            Symbol remapLambda(BoundLambda boundLambda, LambdaSymbol lambda)
+            {
+                var updatedDelegateType = _snapshotManager!.GetUpdatedDelegateTypeForLambda(lambda);
+
+                if (!_remappedSymbols.TryGetValue(lambda.ContainingSymbol, out Symbol? updatedContaining) && updatedDelegateType is null)
+                {
+                    return lambda;
+                }
+
+                LambdaSymbol updatedLambda;
+                if (updatedDelegateType is null)
+                {
+                    Debug.Assert(updatedContaining is object);
+                    var parameterRefKinds = lambda.ParameterRefKinds.IsDefault ? ImmutableArray<RefKind>.Empty : lambda.ParameterRefKinds;
+                    updatedLambda = boundLambda.CreateLambdaSymbol(updatedContaining, lambda.ReturnTypeWithAnnotations, lambda.ParameterTypesWithAnnotations, parameterRefKinds, lambda.RefKind);
+                }
+                else
+                {
+                    Debug.Assert(updatedDelegateType is object);
+                    updatedLambda = boundLambda.CreateLambdaSymbol(updatedDelegateType, updatedContaining ?? lambda.ContainingSymbol);
+                }
+
+                _remappedSymbols.Add(lambda, updatedLambda);
+
+                Debug.Assert(lambda.ParameterCount == updatedLambda.ParameterCount);
+                for (int i = 0; i < lambda.ParameterCount; i++)
+                {
+                    _remappedSymbols.Add(lambda.Parameters[i], updatedLambda.Parameters[i]);
+                }
+
+                return updatedLambda;
+            }
+
+            Symbol remapLocalFunction(LocalFunctionSymbol localFunction)
+            {
+                if (_remappedSymbols.TryGetValue(localFunction, out var updatedSymbol))
+                {
+                    return updatedSymbol;
+                }
+
+                if (!_remappedSymbols.TryGetValue(localFunction.ContainingSymbol, out var updatedContaining))
+                {
+                    return localFunction;
+                }
+
+                var updatedLocalFunction = localFunction.WithContainingSymbol(updatedContaining);
+
+                _remappedSymbols.Add(localFunction, updatedLocalFunction);
+                for (int i = 0; i < localFunction.ParameterCount; i++)
+                {
+                    _remappedSymbols.Add(localFunction.Parameters[i], updatedLocalFunction.Parameters[i]);
+                }
+
+                return updatedLocalFunction;
```

Comment:
>return updatedLocalFunction; [](start = 16, length = 28)

I think we need to add remapping for type parameters as well.

Reply:
Tracked by [https://github.com/dotnet/roslyn/issues/38922](https://github.com/dotnet/roslyn/issues/38922).

---
In reply to: [329753419](https://github.com/dotnet/roslyn/pull/38769#discussion_r329753419) [](ancestors = 329753419)